### ISSUE TITLE
about the loss when training the network

### ISSUE THREAD

**Issue by flyduck** [1adrianb/face-alignment#161]
HI,
first, very thanks for your grate work! I test your model on my own data, its perfect! I want to train my own data, but, I cant understand you said the loss use "sigmoid cross-entropy pixel-wise loss". So I use L2 loss, but the loss decrease so slowly, could you please explain how to create the loss function about "sigmoid cross-entropy pixel-wise loss"? 
Grate thanks!

**Comment by 1adrianb** [comment]
Hi, you can find the loss (BCE) used in pytorch here: https://media.licdn.com/dms/image/C5603AQFIzsLP3W7KVw/profile-displayphoto-shrink_200_200/0?e=1579737600&v=beta&t=q6F-cM26j4w3a2YhhHSYXqWUAzZRjIRFsz-VsRFKUqg

**Comment by 1adrianb** [comment]
Hi @flyduck,
Thanks for your interest! You can find the loss (BCE) used in pytorch here: https://pytorch.org/docs/stable/nn.html#torch.nn.BCELoss

**Comment by flyduck** [comment]
@1adrianb ok, thanks very much. I look your lua code, and I cant find the function "crop" definition, could you please help me tell me where this function? thanks~

**Comment by flyduck** [comment]
@1adrianb I find the function in your another code project, but I have another question, in your code, when get the heat map from gt, the gaussian around the gt point in the 64*64 size, the gaussian maybe could not uniform gaussian? if uniform gaussian there will be some accuracy loss in gt?

**Comment by flyduck** [comment]
@1adrianb hi, I rewrite your code with tensor flow framework, and I want to know if you try the tensor channel with other number, with channel 256 the model is so large~

**Comment by 1adrianb** [comment]
@flyduck you can either reduce the number of HGs from 4 to 2 or decrease the number of channels to something like 128. The accuracy drop should be small

**Comment by flyduck** [comment]
@1adrianb ok, I will be test it, thanks very much!
